cmd/crdvalidator/README.md

```diff
@@ -0,0 +1,96 @@
+
+### How to use
+
+`crdvalidator` will ensure that any `CRD` upgrade is safe as long as it is owned by RukPak. If you would like to disable validation for a `CRD` included in a specific `Bundle`, you will need to explicitly set the `core.rukpak.io/safe-upgrade-validation` annotation to `disabled` within that specific `Bundle`.
+
+```yaml
+apiVersion: apiextensions.k8s.io/v1
+kind: CustomResourceDefinition
+metadata:
+  annotations:
+    core.rukpak.io/safe-upgrade-validation: disabled # <--
+```
+
+### What is protected
+There are three main cases `crdvalidator` is checking against with every specified `CRD` upgrade. These are where a new `CRD`:
+1. Removes a stored version and ensures that removing the stored version does not result in data loss.
+2. Changes a version that old `CRD` had and it must validate existing CRs against the new schema.
+3. Adds a version that old `CRD` does not have. In this case `crdvalidator` checks if the conversion strategy is:
+    - None, then ensures that existing CRs validate with new schema.
+    - Webhook, then allow update (assuming webhook handles conversion correctly)
+
+#### Update the schema that invalidates existing CRs
+Say that you have installed a `CRD` onto the cluster with a Kind of `Sample` that has a single property of `examplearray`. 
+
+```yaml
+apiVersion: apiextensions.k8s.io/v1
+kind: CustomResourceDefinition
+metadata:
+  annotations:
+    controller-gen.kubebuilder.io/version: v0.7.0
+    core.rukpak.io/safe-upgrade-validation: enabled
+  creationTimestamp: null
+  name: templates.combo.io
+#
+# ...
+#
+spec:
+    versions:
+    - name: v1alpha1
+        schema:
+        openAPIV3Schema:
+            properties:
+                examplearray:
+                    description: Examplearray is an array of strings to demonstrate crd validation logic.
+                    type: array
+                    minItems: 1
+                    items:
+                    type: string
+```
+
+With this installed onto the cluster, you go ahead and create a `CR` of a `Sample` onto the cluster. If you were to then update that `CRD`'s schema in such a way that that `CR` you created was invalidated, like so:
+
+```yaml
+#
+# ...
+#
+spec:
+    versions:
+    - name: v1alpha1
+        schema:
+        openAPIV3Schema:
+            properties:
+                examplearray:
+                    description: Examplearray is an array of strings to demonstrate crd validation logic.
+                    type: array
+                    minItems: 100
+                    items:
+                    type: string
+```
+
+
+`crdvalidator` will prevent the update from occurring if the existing `CR` does not have at least 100 items in the `examplearray` field.
+
+## Running locally
+
+`crdvalidator` is included with every default installation of RukPak. To run this locally, simply run the make target with a kind cluster started.
+
+```console
+$ make run
```

Comment:
nit: I think we decided to pull the `$` out of our bash documentation based on previous PRs